### PR TITLE
Added support for OpenTracing

### PR DESCRIPTION
**Pull request by enbohm**
I updated to Wildfly Swarm 2017.10.0 to be able to add the Jaeger fraction. Also, since the 2017.10.0 update, I had to change the datasource definition a bit. In docker-compose.yml I added the service jaegertracing/all-in-one:0.8.0 and published the Jaeger UI on port 8090 where all endpoint requests will be visible.

### CODE REVIEW

File: employees/pom.xml [hunk 1]
```diff
@@ -31,7 +31,7 @@
     <dependencies>
         <dependency>
             <groupId>org.wildfly.swarm</groupId>
-            <artifactId>jaxrs-cdi</artifactId>
+            <artifactId>cdi</artifactId>
```
Comment (arun-gupta) [inline employees/pom.xml:34]: Do we need the complete `cdi` module?

File: employees/pom.xml [hunk 2]
```diff
@@ -47,7 +47,7 @@
         </dependency>
         <dependency>
             <groupId>org.wildfly.swarm</groupId>
-            <artifactId>jaxrs-jaxb</artifactId>
+            <artifactId>jaeger</artifactId>
```
Comment (arun-gupta) [inline employees/pom.xml:50]: Why is `jaxrs-jaxb` module removed? IIRC this was needed for POJO -> XML translation.

File: employees/src/main/resources/project-defaults.yml [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
       ExampleDS:
-        driver-name: com.mysql
+        driver-name: mysql
         connection-url: jdbc:mysql://db:3306/employees
         user-name: mysql
-        password: mysql
\ No newline at end of file
+        password: mysql
+  jaeger:
+    service-name: docker-javaee-service
```
Comment (arun-gupta) [inline employees/src/main/resources/project-defaults.yml]: Can we change the service name to have `jaeger`?

File: readme.adoc [hunk 4]
```diff
@@ -235,3 +235,5 @@ jvm_gc_collection_seconds_count{gc="PS MarkSweep",} 6.0
 jvm_gc_collection_seconds_sum{gc="PS MarkSweep",} 0.565
 ```
 
+== OpenTracing/Jaeger
+The application traces all requests to Jaeger (https://uber.github.io/jaeger/) and the result can be viewed on http://127.0.0.1:8090
```
Comment (arun-gupta) [inline readme.adoc]: Add what would a sample output look like?

File: docker-compose.yml [hunk 5]
```diff
@@ -17,3 +17,12 @@ services:
...
       - db
+  jaeger:
+    image: jaegertracing/all-in-one:0.8.0
+    ports:
+      - 8090:16686
+      - 5775:5775/udp
+      - 5778:5778
+      - 6831:6831/udp
+      - 6832:6832/udp
+      - 14268:14268
```
Comment (arun-gupta) [inline docker-compose.yml]: Do all the ports need to be published? Is there a minimal set of ports that can be used?

**Review by arun-gupta — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 5 above)
Comment (enbohm) [reply docker-compose.yml]: Will update to use fewer ports - if other ports are needed please refer to http://jaeger.readthedocs.io/en/latest/getting_started/ which I initially got the port mappings from.

File: employees/pom.xml (hunk 1 above)
Comment (enbohm) [reply employees/pom.xml:34]: javax.enterprise.context.ApplicationScoped in not available in jaxrs-cdi fraction anymore and hence I added the cdi-fraction instead

File: employees/pom.xml (hunk 2 above)
Comment (enbohm) [reply employees/pom.xml:50]: I think JAXB is now available via the jaxrs fraction (POJO->XML is working)

File: readme.adoc (hunk 4 above)
Comment (enbohm) [reply readme.adoc]: Will add a small screenshot of the Jaeger UI with some tracing examples

**Comment by arun-gupta** [comment]
@enbohm Started the application using the command:

`docker stack deploy --compose-file=docker-compose.yml webapp`

Jaeger UI is showing the following output:

screen shot 2017-10-17 at 8 33 15 pm

Is there any further configuration required?

**Comment by enbohm** [comment]
You need to trigger one request before anything is visible in Jeager. Try accessing http://127.0.0.1:8080/resources/employees - this will make the first trace visible in Jeager. The service 'jeager' should then be visible in the dropdown below "Service".

**Comment by arun-gupta** [comment]
Already did that:

```
744  curl http://localhost:8080/resources/employees
  746  curl http://localhost:8080/resources/employees/1
  747  curl http://localhost:8080/resources/employees/2
  748  curl http://127.0.0.1:8080/resources/employees/2
```

But still getting as shown:

screen shot 2017-10-17 at 8 49 27 pm

**Comment by enbohm** [comment]
Are you using correct image? If I'm pulling image: arungupta/docker-javaee:dockerconeu17 it is not working but if I use image: arungupta/docker-javaee:latest (after I've run maven clean package -Pdocker) it works. It should also be visible in the service logs (incl. Jeager fraction settings)
servicelogs

**Comment by arun-gupta** [comment]
Building the new image works, looks good now, thanks!

**Comment by enbohm** [comment]
Great! If you want to use OpenTracing with EJBs later on this article might be of interest (I found it quite useful) https://dzone.com/articles/opentracing-ejb-instrumentation-on-wildfly-swarm

Cheers,
Andreas

**Comment by arun-gupta** [comment]
Thanks!

Have not used EJBs for a few years now ;)

**Comment by enbohm** [comment]
haha - well in case you'll ever start using EJBs again then you will know how to instrument them with Jeager ;)

**Pull request merged**